extensions/extensionservice/org.eclipse.smarthome.extensionservice.marketplace/src/main/java/org/eclipse/smarthome/extensionservice/marketplace/internal/BindingExtensionHandler.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.extensionservice.marketplace.internal;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.LineIterator;
+import org.eclipse.smarthome.extensionservice.marketplace.MarketplaceExtension;
+import org.eclipse.smarthome.extensionservice.marketplace.MarketplaceExtensionHandler;
+import org.eclipse.smarthome.extensionservice.marketplace.MarketplaceHandlerException;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.BundleException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link MarketplaceExtensionHandler} implementation, which handles bindings as jar files (OSGi bundles) and installs
+ * them through the standard OSGi bundle installation mechanism.
+ * The information, which installed bundle corresponds to which extension is written to a file in the bundle's data
+ * store. It is therefore wiped together with the bundles upon an OSGi "clean".
+ * We might want to move this class into a separate bundle in future, when we add support for further extension types.
```

Comment:
Wouldn't it be already a good time as we have a different one for the automation one?